### ISSUE TITLE
Add option for disabling auth check on OPTIONS request

### ISSUE THREAD

**Issue by Debetux** [fastify/fastify-bearer-auth#56]
## 🚀 Feature Proposal

Add a parameter to disable the auth check on `OPTIONS` requests.

## Motivation

An OPTIONS request (e.g. in the case of checking CORS) returns a 401. Since including auth header in OPTIONS request seems to be a bad practice, would it be useful to disable auth header check on OPTIONS request? Or am I missing an obvious way of doing this?

## Example

```
fastify.register(bearerAuthPlugin, {
    keys,
    methods: ['POST', 'GET', 'DELETE', 'PUT']
})
```

P.S. : if this is good idea, I am ready to make a pull request

**Comment by Eomm** [comment]
You could archive this by:

```js
fastify.register((instance, opts, next) => {
  instance.register(bearerAuthPlugin, {
      keys
  })
  instance.get('/foo', (req, reply) => {
    reply.send({authenticated: true})
  })
  next()
})
fastify.options('/*', (request, reply)=>{ reply.send('ok') })
```

If you would like to send a PR I would add something generic like `apply or skip filter` that is a function the user may customized and it must return a boolean for every request.

**Comment by mcollina** [comment]
I prefer not to add a skip here. You can use fastify-cors@4, or wait for https://github.com/fastify/fastify-cors/pull/103 to land.

**Comment by jsumners** [comment]
> I prefer not to add a skip here. You can use fastify-cors@4, or wait for [fastify/fastify-cors#103](https://github.com/fastify/fastify-cors/pull/103) to land.

Correct. This plugin's concern should remain authorization and authorization only. Handling the CORS spec is out of scope.

**Comment by Debetux** [comment]
I made a custom fix in my code, while waiting for fastify/fastify-cors#103 to land.
Thanks for all the responses!

**Comment by codyzu** [comment]
@Debetux this should be resolved with the 5.1.0 release of fastify-cors, returning the pre 5.0.0 behavior and removing the need to not authenticate OPTIONS requests. With the default options, fastify-cors should now automatically reply to preflight requests before your authentication plugin.
